persistence/trees/trees.go

```diff
@@ -0,0 +1,574 @@
+func (t *treeStore) getActorsUpdated(
+	pgtx pgx.Tx,
+	actorType coreTypes.ActorType,
+	height uint64,
+) ([]*coreTypes.Actor, error) {
+	actorSchema, ok := actorTypeToSchemaName[actorType]
+	if !ok {
+		return nil, fmt.Errorf("no schema found for actor type: %s", actorType)
+	}
+
+	// TECHDEBT (ISSUE #813): Avoid this cast to int64
+	// https://github.com/pokt-network/pocket/issues/813
+	query := actorSchema.GetUpdatedAtHeightQuery(int64(height))
+	rows, err := pgtx.Query(context.TODO(), query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	addrs := make([][]byte, 0)
+	for rows.Next() {
+		var addr string
+		if err := rows.Scan(&addr); err != nil {
+			return nil, err
+		}
+		addrBz, err := hex.DecodeString(addr)
+		if err != nil {
+			return nil, err
+		}
+		addrs = append(addrs, addrBz)
+	}
+
+	actors := make([]*coreTypes.Actor, len(addrs))
+	for i, addr := range addrs {
+		// TECHDEBT #XXX: Avoid this cast to int64
+		actor, err := t.getActor(pgtx, actorSchema, addr, int64(height))
+		if err != nil {
+			return nil, err
+		}
+		actors[i] = actor
+	}
+	rows.Close()
+
+	return actors, nil
+}
+
+func (t *treeStore) getAccountsUpdated(
+	pgtx pgx.Tx,
+	acctType ptypes.ProtocolAccountSchema,
+	height uint64,
+) ([]*coreTypes.Account, error) {
+	accounts := []*coreTypes.Account{}
+
+	// TECHDEBT (ISSUE #813): Avoid this cast to int64
+	// https://github.com/pokt-network/pocket/issues/813
+	query := acctType.GetAccountsUpdatedAtHeightQuery(int64(height))
+	rows, err := pgtx.Query(context.TODO(), query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		acc := new(coreTypes.Account)
+		if err := rows.Scan(&acc.Address, &acc.Amount); err != nil {
+			return nil, err
+		}
+		accounts = append(accounts, acc)
+	}
+
+	return accounts, nil
+}
+
+func (t *treeStore) getTransactions(txi indexer.TxIndexer, height uint64) ([]*coreTypes.IndexedTransaction, error) {
+	// TECHDEBT (ISSUE #813): Avoid this cast to int64
+	// https://github.com/pokt-network/pocket/issues/813
+	indexedTxs, err := txi.GetByHeight(int64(height), false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get transactions by height: %w", err)
+	}
+	return indexedTxs, nil
+}
+
+// getPools returns the pools updated at the given height
+func (t *treeStore) getPools(pgtx pgx.Tx, height uint64) ([]*coreTypes.Account, error) {
+	pools, err := t.getAccountsUpdated(pgtx, ptypes.Pool, height)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pools: %w", err)
+	}
+	return pools, nil
+}
+
+// getAccounts returns the list of accounts updated at the provided height
+func (t *treeStore) getAccounts(pgtx pgx.Tx, height uint64) ([]*coreTypes.Account, error) {
+	accounts, err := t.getAccountsUpdated(pgtx, ptypes.Account, height)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get accounts: %w", err)
+	}
+	return accounts, nil
+}
+
+func (t *treeStore) getFlags(pgtx pgx.Tx, height uint64) ([]*coreTypes.Flag, error) {
```

Comment:
Prior to this change, we were using a helper:

![Screenshot 2023-06-14 at 1 24 27 PM](https://github.com/pokt-network/pocket/assets/1892194/e69fd525-4f82-429c-a9e8-443b210337e3)

I'm not a fan of as it lets SQL spill over into new pats of the code that were otherwise blind to it

![Screenshot 2023-06-14 at 1 24 44 PM](https://github.com/pokt-network/pocket/assets/1892194/ddd56718-5fda-4d58-80c5-30351c08c9f4)

I feel strongly that we either revert or encapsulate the logic elsewhere, or please explain the logic/reasoning here.

Reply:
So, there's an import cycle error that arises if we try to use the existing SQL function, which is why I had to have the SQL handling in the trees package. I think one option would be to make a sibling package in `persistence/` that abstracts away the SQL handling and then `trees` package could import it. Do you have a better idea? I didn't think it was too ugly to have the SQL handling in this package but I can understand wanting to limit that handling and scope.

Reply:
```
So, there's an import cycle error that arises if we try to use the existing SQL function, 
```

Can you show/explain what the import cycle is?

```
think one option would be to make a sibling package in persistence/ that abstracts away the SQL handling and then trees package could import it.
```

Sounds reasonable to me. Though I do want to understand the above first.


```
 I didn't think it was too ugly to have the SQL handling in this package but I can understand wanting to limit that handling and scope.
 ```
 
 Yea, I'd like to avoid this.

Reply:
@Olshansk okay I pushed up a commit that places the functions in `persistence/sql` package and I think we should TECHDEBT refactoring the other PersistenceContext helpers that handle SQL into that packge. How does that sound?

Reply:
Let's do it. Wealready have 2 comments on main w.r.t to doing this

![Screenshot 2023-06-16 at 4 18 32 PM](https://github.com/pokt-network/pocket/assets/1892194/0c44669b-1e1b-47c4-8bd7-e13c8f859e65)

Reply:
Is there an existing issue for this? If not, I'll make one for it. Otherwise, I'll consider these comments sufficient.